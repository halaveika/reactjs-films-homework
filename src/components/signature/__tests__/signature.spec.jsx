import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Signature from '..';

it('Signature component renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Signature name="Aliaksandr Halaveika" />);
  const result = renderer.getRenderOutput();

  expect(result.type).toBe('h2');
  expect(result.props.children).toEqual('Aliaksandr Halaveika');
  expect(renderer).toMatchSnapshot();
});
