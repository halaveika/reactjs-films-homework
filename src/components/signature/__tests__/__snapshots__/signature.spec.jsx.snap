// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signature component renders correctly 1`] = `
ReactShallowRenderer {
  "_context": Object {},
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useDeferredValue": [Function],
    "useEffect": [Function],
    "useImperativeHandle": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useResponder": [Function],
    "useState": [Function],
    "useTransition": [Function],
  },
  "_element": <Signature
    name="Aliaksandr Halaveika"
  />,
  "_firstWorkInProgressHook": null,
  "_forcedUpdate": false,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": <h2>
    Aliaksandr Halaveika
  </h2>,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;
